src/main/scala/is/hail/variant/MatrixTable.scala

```diff
@@ -1231,7 +1231,7 @@ class MatrixTable(val hc: HailContext, val ast: MatrixIR) {
     }
 
     assert(finalNames.areDistinct())
-    val touchesKeys = !rowKey.forall(topLevelFields.contains)
+    val touchesKeys = rowKey.filter(topLevelFields.contains) != rowKey
```

Comment:
this is two linear passes (one to filter and one for equality check) it seems like `rowKey.exists(topLevelFields.contains)` is the most efficient way to do this that doesn't require creating sets.